core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/domain/solution/cloner/DeepCloningUtils.java

```diff
@@ -72,14 +77,13 @@ public final class DeepCloningUtils {
             HardSoftScore.class, HardSoftLongScore.class, HardSoftBigDecimalScore.class,
             HardMediumSoftScore.class, HardMediumSoftLongScore.class, HardMediumSoftBigDecimalScore.class,
             BendableScore.class, BendableLongScore.class, BendableBigDecimalScore.class);
+
+    private final Map<Pair<Field, Class<?>>, Boolean> fieldDeepClonedMemoization = new ConcurrentHashMap<>();
```

Comment:
The `ConcurrentMemoization` had a purpose there; see https://issues.redhat.com/browse/PLANNER-706.

The trick is that `get()` is faster than `computeIfAbsent()` if updates to the map are rare. Internally, `computeIfAbsent()` has to do more locking.